### PR TITLE
Channel#delete_overwrite

### PR DESCRIPTION
**Pull request by greenbigfrog**

### CODE REVIEW

File: lib/discordrb/data.rb [hunk 1]
```diff
@@ -1230,6 +1231,14 @@ def define_overwrite(thing, allow, deny)
       API::Channel.update_permission(@bot.token, @id, thing.id, allow_bits, deny_bits, type)
     end
 
+    # Deletes a permission overwrite for this channel
+    # @param thing [Member, User, Role, Profile, Reciepient]
```
Comment (meew0) [inline lib/discordrb/data.rb]: You misspelled `Recipient` here. Also, you should write a short comment what `thing` should be. I'd also use a more descriptive name (perhaps `target`)?

File: lib/discordrb/data.rb [hunk 2]
```diff
@@ -1230,6 +1231,14 @@ def define_overwrite(thing, allow, deny)
...
+    def delete_overwrite(thing)
+      raise 'Tried deleting a overwrite for a invalid thing' unless thing.is_a?(Member) || thing.is_a?(User) || thing.is_a?(Role) || thing.is_a(Profile) || thing.is_a(Receipient)
```
Comment (meew0) [inline lib/discordrb/data.rb]: `Recipient` is misspelled here too

File: lib/discordrb/data.rb [hunk 3]
```diff
@@ -1232,11 +1232,15 @@ def define_overwrite(thing, allow, deny)
...
     # Deletes a permission overwrite for this channel
-    # @param thing [Member, User, Role, Profile, Reciepient]
-    def delete_overwrite(thing)
-      raise 'Tried deleting a overwrite for a invalid thing' unless thing.is_a?(Member) || thing.is_a?(User) || thing.is_a?(Role) || thing.is_a(Profile) || thing.is_a(Receipient)
+    # @param target [Member, User, Role, Profile, Recipient] What permission overwrite to delete
+    def delete_overwrite(target)
+      raise 'Tried deleting a overwrite for a invalid target' unless target.is_a?(Member) || target.is_a?(User) || target.is_a?(Role) || target.is_a?(Profile) || target.is_a?(Recipient) || target.respond_to?(:resolve_id)
 
-      API::Channel.delete_permission(@bot.token, @id, thing.id)
+      if target.respond_to?(:resolve_id)
```
Comment (meew0) [inline lib/discordrb/data.rb]: The data classes also implement `resolve_id`, so you can simplify that if statement into just that.

File: lib/discordrb/data.rb [hunk 4]
```diff
@@ -1232,11 +1232,15 @@ def define_overwrite(thing, allow, deny)
     end
 
     # Deletes a permission overwrite for this channel
-    # @param thing [Member, User, Role, Profile, Reciepient]
-    def delete_overwrite(thing)
-      raise 'Tried deleting a overwrite for a invalid thing' unless thing.is_a?(Member) || thing.is_a?(User) || thing.is_a?(Role) || thing.is_a(Profile) || thing.is_a(Receipient)
+    # @param target [Member, User, Role, Profile, Recipient] What permission overwrite to delete
+    def delete_overwrite(target)
+      raise 'Tried deleting a overwrite for a invalid target' unless target.is_a?(Member) || target.is_a?(User) || target.is_a?(Role) || target.is_a?(Profile) || target.is_a?(Recipient) || target.respond_to?(:resolve_id)
```
Comment (meew0) [inline lib/discordrb/data.rb]: "an invalid target"

File: lib/discordrb/data.rb (hunk 3 above)
Comment (greenbigfrog) [inline lib/discordrb/data.rb]: TBH I expected sth like this

File: lib/discordrb/data.rb [hunk 5]
```diff
@@ -1230,6 +1231,14 @@ def define_overwrite(thing, allow, deny)
       API::Channel.update_permission(@bot.token, @id, thing.id, allow_bits, deny_bits, type)
     end
 
+    # Deletes a permission overwrite for this channel
+    # @param target [Member, User, Role, Profile, Recipient] What permission overwrite to delete
```
Comment (meew0) [inline lib/discordrb/data.rb]: One last thing; the type parameter here should also mention `#resolve_id`.

Comment (greenbigfrog) [inline lib/discordrb/data.rb]: why do I always miss stuff like this...

**Pull request merged**